### PR TITLE
Add new Insurance Claim Automation demo

### PR DESCRIPTION
**Pull request by AlbertDavletov**

### CODE REVIEW

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+            }
+            catch (Exception ex)
+            {
+                await Util.GenericApiCallExceptionHandler(ex, "Failure recognizing form");
+            }
+
+            return null;
+        }
+
+        private async Task<ImagePrediction> AnalyzeObjectAsync(Guid modelId)
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: AnalyzeProductImageAsync

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+                    result = await CustomVisionServiceHelper.PredictImageUrlWithRetryAsync(predictionApi, modelId, new ImageUrl(img.ImageUrl), latestTrainedIteraction.Id);
+                }
+                else if (img?.GetImageStreamCallback != null)
+                {
+                    result = await CustomVisionServiceHelper.PredictImageWithRetryAsync(predictionApi, modelId, img.GetImageStreamCallback, latestTrainedIteraction.Id);
+                }
+            }
+            catch (Exception ex)
+            {
+                await Util.GenericApiCallExceptionHandler(ex, "Failure detecting object");
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Custom Vision error analyzing product image

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+            }
+            finally
+            {
+                this.dataGrid.Tag = null;
+                this.dataGrid.SelectedItem = null;
+            }
+        }
+    }
+
+    public enum InputViewState
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Can you move these secondary classes/enums to a DataModel.cs file, under a DataModel folder?

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimData.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+// 
+
+using Microsoft.Azure.CognitiveServices.FormRecognizer.Models;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.Foundation;
+
+namespace IntelligentKioskSample.Views.InsuranceClaimAutomation
+{
+    public class TokenOverlayInfo
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimData.cs]: Move this file under DataModel folder too

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimDataLoader.cs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.Storage;
+
+namespace IntelligentKioskSample.Views.InsuranceClaimAutomation
+{
+    public class InsuranceClaimDataLoader
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimDataLoader.cs]: Move this file under DataModel folder

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,643 @@
...
+using Windows.Storage;
+using Windows.UI.Popups;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Navigation;
+
+namespace IntelligentKioskSample.Views.InsuranceClaimAutomation
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Replace this default description

File: Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+            <Pivot Grid.Row="1" Background="{StaticResource PivotBackgroundColor}">
+                <PivotItem Margin="17,13,17,19">
+                    <PivotItem.Header>
+                        <TextBlock Text="Product image" FontSize="15" />
+                    </PivotItem.Header>
+
+                    <Grid Background="{StaticResource PivotItemBackgroundColor}">
+                        <Grid HorizontalAlignment="Center">
+                            <ctl:ImageWithFaceBorderUserControl x:Name="productImageControl" />
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml]: Let's just use a Image control for this.

File: Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+                <PivotItem Margin="17,13,17,19">
+                    <PivotItem.Header>
+                        <TextBlock Text="Form image" FontSize="15" />
+                    </PivotItem.Header>
+
+                    <Grid Background="{StaticResource PivotItemBackgroundColor}">
+                        <ScrollViewer>
+                            <Grid HorizontalAlignment="Stretch">
+                                <ctl:ImageWithFaceBorderUserControl x:Name="formImageControl" HorizontalAlignment="Stretch"/>
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml]: Same as above

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+                                           ItemsSource="{Binding DataGridCollection}"
+                                           SelectionChanged="DataGridSelectionChanged">
+                            <controls:DataGrid.Columns>
+                                <controls:DataGridTextColumn Header="Claim Id" Binding="{Binding ClaimId}"/>
+
+                                <controls:DataGridTemplateColumn Header="Product Image">
+                                    <controls:DataGridTemplateColumn.CellTemplate>
+                                        <DataTemplate>
+                                            <Grid>
+                                                <ctl:ImageWithFaceBorderUserControl Width="100" Height="70" DataContext="{Binding ProductImage}"/>
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml]: Use standard Image control here too

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+                                            </Grid>
+                                        </DataTemplate>
+                                    </controls:DataGridTemplateColumn.CellTemplate>
+                                </controls:DataGridTemplateColumn>
+
+                                <controls:DataGridTemplateColumn Header="Form Image">
+                                    <controls:DataGridTemplateColumn.CellTemplate>
+                                        <DataTemplate>
+                                            <Grid Width="100" Height="70">
+                                                <ctl:ImageWithFaceBorderUserControl DataContext="{Binding FormImage}" Visibility="{Binding IsFormImage, Converter={StaticResource booleanToVisibilityConverter}}"/>
```
Comment (CarlosP-MS) [inline Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml]: Same as above

**Review by CarlosP-MS — CHANGES_REQUESTED** [review]

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs (hunk 1 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs (hunk 2 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs (hunk 3 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimData.cs (hunk 4 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimData.cs]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimDataLoader.cs (hunk 5 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimDataLoader.cs]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs (hunk 6 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml.cs]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml (hunk 7 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml (hunk 8 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/DetailsViewControl.xaml]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml (hunk 9 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml]: Fixed.

File: Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml (hunk 10 above)
Comment (AlbertDavletov) [reply Kiosk/Views/InsuranceClaimAutomation/InsuranceClaimAutomation.xaml]: Fixed.

**Review by CarlosP-MS — APPROVED** [review]

**Pull request merged**